### PR TITLE
[WIP] Remove RAFT memory management

### PR DESCRIPTION
**Pull request by viclafargue**
Answers #308.
Requires the appropriate changes in `cuML` and `cuGraph` before merging.

### CODE REVIEW

**Comment by cjnolet** [comment]
@viclafargue, it looks like there's just a couple includes for `allocator` remaining in the code and this is ready to go?

**Comment by viclafargue** [comment]
Postponing to 22.04, as discussed offline.

**Comment by cjnolet** [comment]
rerun tests

File: cpp/include/raft/mr/host/allocator.hpp [hunk 1]
```diff
@@ -1,61 +0,0 @@
...
- * further to the ones listed in `Allocator`:
- * - Allocations don't need to be zero copy accessible form a device.
- */
-class allocator : public base_allocator {
-};
-
-/** Default cudaMallocHost/cudaFreeHost based host allocator */
-class default_allocator : public allocator {
- public:
-  void* allocate(std::size_t n, cudaStream_t stream) override
```
Comment (cjnolet) [inline cpp/include/raft/mr/host/allocator.hpp:44]: I had tried to remove this at one point and realized cugraph is still using it. We should also open a PR in cugraph to move the use of host allocator/buffer to something equivalent.

Comment (seunghwak) [reply cpp/include/raft/mr/host/allocator.hpp:44]: I just searched cugraph/cpp and could not find code calling this. We need to double check but we may have removed the code using this function.

Comment (cjnolet) [reply cpp/include/raft/mr/host/allocator.hpp:44]: @seunghwak I think you are right, I'm no longer seeing it in cugraph either.

**Review by cjnolet — APPROVED** [review]

**Comment by cjnolet** [comment]
@gpucibot merge

**Pull request merged**